scipy/optimize/minpack.py

```diff
@@ -145,12 +144,19 @@ def fsolve(func, x0, args=(), fprime=None, full_output=0,
         info['fvec'] = res['fun']
         return x, info, res['status'], res['message']
     else:
+        status = res['status']
+        msg = res['message']
+        if status != 1:
+            if status in [2, 3, 4, 5]:
+                warnings.warn(msg, RuntimeWarning)
+            else:
+                raise TypeError(msg)
```

Comment:
Though the above will change the behavior of the code from the original: if an unknown status gets returned it will only give a warning instead of raising an exception. Perhaps the following is clear yet preserves the behavior:
```python
if status == 0:
     raise TypeError(msg)
elif status == 1:
    pass
elif status in [2, 3, 4, 5]:
    warnings.warn(msg, RuntimeWarning)
else:
    raise TypeError(msg)
```